### PR TITLE
Add map fragment as compose element

### PR DESCRIPTION
**Pull request by petter-mfn**
- Switch view model to shared with list view, so that the same search field is used
- Switch between map and list with toggle button
- Add location permission management to GPS toggle FAB
- Remove location permission from observation map fragment

### CODE REVIEW

File: app/src/main/java/berlin/mfn/naturblick/ui/composable/ToggleGPSFab.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+@Composable
+fun ToggleGPSFab(
+    locationEnabled: Boolean,
+    modifier: Modifier = Modifier,
+    onToggle: (Boolean) -> Unit
+) {
+    val permissionRequest = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestMultiplePermissions(),
+        onResult = { permissions ->
+            if (permissions.values.any { it }) {
```
Comment (j-ebbighausen-mfn) [inline app/src/main/java/berlin/mfn/naturblick/ui/composable/ToggleGPSFab.kt:29]: Are you sure that `any` is correct? I would think that `all` is correct.

Comment (j-ebbighausen-mfn) [reply app/src/main/java/berlin/mfn/naturblick/ui/composable/ToggleGPSFab.kt:29]: i see, you are asking for `COARSE` respectively `FINE` location and one is sufficient.

File: app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookMapFragment.kt [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                    map.addOnMapClickListener {
+                        mapView.viewAnnotationManager.removeAllViewAnnotations()
+                        false
+                    }
+                    pointAnnotationManager.deleteAll()
+                    mapView.viewAnnotationManager.removeAllViewAnnotations()
+                    val annotations: List<PointAnnotationOptions> =
+                        observations.mapNotNull { observation ->
+                            observation.coords?.let { coords ->
+                                observation.species?.group?.let { group ->
```
Comment (j-ebbighausen-mfn) [inline app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookMapFragment.kt:95]: if no species is set you should set the unknown `?` marker.

**Comment by j-ebbighausen-mfn** [comment]
"Switch between map and list with toggle button" 
does it mean to use "map button" and "map button filled"? Because then, it's not yet done.

**Comment by petter-mfn** [comment]
Map button filled is iOS, in android the button should update the icon instead.

File: app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookMapFragment.kt (hunk 2 above)
Comment (petter-mfn) [reply app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookMapFragment.kt:95]: Good catch!

**Review by j-ebbighausen-mfn — APPROVED** [review]

**Pull request merged**